transport.go

```diff
@@ -9,14 +9,18 @@ import (
 )
 
 const (
-	sshControlPath    = "~/.ssh/judo-control-%C"
-	sshControlPathOpt = "-o ControlPath " + sshControlPath
+	sshControlPath        = "~/.ssh/judo-control-%C"
+	sshControlPathOpt     = "-o ControlPath " + sshControlPath
+	sshControlMasterNoOpt = "-o ControlMaster no"
 )
 
 func (host *Host) pushFiles(job *Job,
 	fnameLocal string, fnameRemote string) (err error) {
-	var remote = fmt.Sprintf("[%s]:%s", host.Name, fnameRemote)
-	proc, err := NewProc("scp", sshControlPathOpt, "-r", fnameLocal, remote)
+	var remote = fmt.Sprintf("%s:%s", host.Name, fnameRemote)
```

Comment:
The brackets have been explicitly added to allow specifying IPv6 addresses as host names. The _unintended_ side-effect of preventing adding a username here is actually in line if the _intended_ goal of **not** supporting every possible flag and feature of `ssh(1)`/`scp(1)` - you should be able to handle every possible edge case in your `ssh_config`.

Reply:
I see. 3 year mystery solved. :)